modern/org/commcare/modern/database/DatabaseHelper.java

```diff
@@ -76,6 +76,10 @@
 
             set = true;
         }
+        // we couldn't match any of the fields to our columns
+        if(!set){
+            throw new IllegalArgumentException();
```

Comment:
`throw new IllegalArgumentException()` -> `throw new IllegalArgumentException("Unable to match provided fields with columns.")` or something along those lines